scripts/travis/kafka-integration-test.sh

```diff
@@ -4,6 +4,14 @@ set -e
 
 docker pull spotify/kafka
 CID=$(docker run -d -p 2181:2181 -p 9092:9092 --env ADVERTISED_HOST=localhost --env ADVERTISED_PORT=9092 --rm spotify/kafka)
+
+# Guarantees no matter what happens, docker will remove the instance at the end.
+trap 'docker rm -f $CID 2>/dev/null' EXIT INT TERM
+
 export STORAGE=kafka
+while true; do
+    if nc -z localhost 9092; then
```

Comment:
Does it fail immediately if the port is not up, or does it block for a few ms/seconds?

Reply:
Looks like it blocks forever until port is available. The build still failed though.

Reply:
It claims to "Specifies that nc should just scan for listening daemons, without sending any data to them." (https://linux.die.net/man/1/nc). I was checking source to see details, but will get back to you on that. Meanwhile, will investigate build issue.

Reply:
Build didn't fail. @black-adder canceled every test but the kafka test and it has succeeded at least once. He said he would retry it a few more times to guarantee it works.

Reply:
From a practical viewpoint, when I run `nc -z localhost 9092` on my MacBook, and Kafka is not running, the command returns within 10 ms (using `time` on 100 calls in a row gives about 500 ms).